From f94178077d1128326374e26e825a2c5fff20c350 Mon Sep 17 00:00:00 2001
From: Ben Pfaff <blp@ovn.org>
Date: Wed, 10 Oct 2018 13:35:00 -0700
Subject: [PATCH] bond: Honor updelay and downdelay when LACP is in use.

Since OVS added LACP support back in 2011, bonds have ignored the updelay
and downdelay values for bonds with configured LACP.  The reason is not
clear, but at least one user needs support in this case, so this commit
enables it.

Reported-at: https://mail.openvswitch.org/pipermail/ovs-discuss/2018-October/047490.html
Reported-by: Daniel Leaberry <dleaberry@purestorage.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
Acked-by: Justin Pettit <jpettit@ovn.org>
diff --git a/ofproto/bond.c b/ofproto/bond.c
index b3e9f4dc0..0bdfdf118 100644
--- a/ofproto/bond.c
+++ b/ofproto/bond.c
@@ -1704,8 +1704,7 @@ bond_link_status_update(struct bond_slave *slave)
             VLOG_INFO_RL(&rl, "interface %s: will not be %s",
                          slave->name, up ? "disabled" : "enabled");
         } else {
-            int delay = (bond->lacp_status != LACP_DISABLED ? 0
-                         : up ? bond->updelay : bond->downdelay);
+            int delay = up ? bond->updelay : bond->downdelay;
             slave->delay_expires = time_msec() + delay;
             if (delay) {
                 VLOG_INFO_RL(&rl, "interface %s: will be %s if it stays %s "
